Fix example Time Pattern Value

garritye opened the pull request:
Original sample code was rejected by Home Assistant.  When I change code to remove the surrounding quotes, Home Assistant saw this as valid.  My changed code works properly.  I didn't check the other examples.

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ x ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [  x ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

garritye committed:
Fix example Time Pattern Value

Original sample code was rejected by Home Assistant.  When I change code to remove the surrounding quotes, Home Assistant saw this as valid.  My changed code works properly.  I didn't check the other examples.

frenck commented on source/_docs/automation/trigger.markdown:
```diff
@@ -763,7 +763,7 @@ automation 3:
     - platform: time_pattern
       # You can also match on interval. This will match every 5 minutes
-      minutes: "/5"
+      minutes: /5
```
This change is incorrect. Time pattern is a string and, therefore, should be enquoted.

garritye commented:
Hi Frank,

First of all, I thank you for authoring this triggering function.  It's
exactly what I needed.  However, I still think that quoting creates a
problem, as shown below.

It does not allow me to save when I quote this, either in single or double
quotes.  These are the messages I receive:
[image: image.png]
and...
[image: image.png]

But, when I remove the quotes, it works fine.  This is the same screen
after changing and saving.

[image: image.png]

Is there something else I can use to help document this?

Thank you. I appreciate your prompt attention.

Ed Garrity.


On Mon, Jan 16, 2023 at 2:35 AM Franck Nijhof ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In source/_docs/automation/trigger.markdown
> <https://github.com/home-assistant/home-assistant.io/pull/25777#discussion_r1070912863>
> :
>
> > @@ -763,7 +763,7 @@ automation 3:
>    trigger:
>      - platform: time_pattern
>        # You can also match on interval. This will match every 5 minutes
> -      minutes: "/5"
> +      minutes: /5
>
> This change is incorrect. Time pattern is a string and, therefore, should
> be enquoted.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant.io/pull/25777#pullrequestreview-1249673684>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5ITRYOFBX6NT5XOBPXPI4DWST24NANCNFSM6AAAAAAT4HSPMY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

frenck commented:
> I thank you for authoring this triggering function.

I didn't create it, sorry.

> However, I still think that quoting creates a problem, as shown below.

Nothing is visible, as you used email to respond. Please don't do that. It doesn't work (well just partly).

../Frenck

garritye commented:
Frenck, Thanks for the reply.

The issue is with the visual editor it seems.
When I enter `"/5"` as the time pattern in the visual editor, then look at the YAML it generates, it's attempting to escape the quote.
It generates the following YAML code:

```
  - platform: time_pattern
    hours: "\"/5\""
```

This produces the following error that's shown on screen: `Message malformed: invalid time_pattern value for dictionary value @ data['hours']`.

When I don't include the quotes in the visual editor, it produces the following YAML:

```
  - platform: time_pattern
    hours: /5
```

And this does work.

The documentation looks like it's for YAML, so it's not wrong.
I've changed my fork to restore the quotes, and added a note that in the visual editor, quotes should not be added.

Is that the best solution, or do you think I should drop the PR and open an issue regarding the visual editor generating bad code if quotes are used?

Thanks and have a great day !

frenck commented:
> When I enter "/5" as the time pattern in the visual editor

Yes? That is expected. It will make a string for you.

Sounds like you want to report an issue with the Home Assistant frontend. If so, please raise an issue in the frontend issue tracker. The change you are suggesting in this PR is, however, incorrect.

../Frenck

garritye commented:
I agree.  Thank you.

Have a great day!

Ed

On Tue, Jan 17, 2023 at 3:38 AM Franck Nijhof ***@***.***>
wrote:

> When I enter "/5" as the time pattern in the visual editor
>
> Yes? That is expected. It will make a string for you.
>
> Sounds like you want to report an issue with the Home Assistant frontend.
> If so, please raise an issue in the frontend issue tracker. The change you
> are suggesting in this PR is, however, incorrect.
>
> ../Frenck
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant.io/pull/25777#issuecomment-1385026904>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A5ITRYL4RZ7S3GRAFHLH7ATWSZLBPANCNFSM6AAAAAAT4HSPMY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
